### ISSUE TITLE
More parameters ?

### ISSUE THREAD

**Issue by jesuisnicolasdavid** [maxpumperla/hyperas#6]
Hello Max, are you planning or is it even possible to run a keras model through hyperas with multiple choice of activation function or multiple optimizer ? Thanks!

**Comment by maxpumperla** [comment]
Sure, ```choice``` is your friend:

```{python}
model.add(Activation( {{choice('relu', 'sigmoid', 'tanh')}} ))
 model.compile(loss='categorical_crossentropy', optimizer= {{choice('sgd', 'adam')}} )
```
See section 2.1. here https://github.com/hyperopt/hyperopt/wiki/FMin for more on what you can do. This whole wiki is a recommended read.

And just in case you may want to ask, yes, you can also create more complex things with choices, like conditionally adding layers etc.:

```{python}
if {{choice('four', 'five')}} == 'four':
    model.add(AdditionalLayer)
```

**Comment by maxpumperla** [comment]
Essentially, strings or numeric types can be chosen and numeric types can be sampled from various distributions. Theoretically, you could even choose between different layer types, though this will look very messy and is not recommended.

```{python}
model.add( {{ choice(Activation('relu'), Activation('sigmoid')) }} )
```

**Comment by jesuisnicolasdavid** [comment]
Well That's nice, maybe you should add those to your example, i found them very powerfull features! Have a good day Max.

**Comment by maxpumperla** [comment]
Thanks, you too!

**Comment by jesuisnicolasdavid** [comment]
Last quick one : Is it possible to do this so we can deactivate the layer if the score does not improve with it?
` model.add(Activation( {{choice(['relu', 'sigmoid', 'tanh', 'None'])}} ))`

**Comment by maxpumperla** [comment]
There's at least two ways to do this: 
- with a conditional like the if statement above. That is the better one.
- with Activation('linear'), which doesn't do anything, but only works for your activation example

**Comment by maxpumperla** [comment]
Check out the new README and the corresponding complex.py example. It should contain most of what we discussed.

Thanks for being so active, this helped me push a few things sooner than I thought.

**Comment by jesuisnicolasdavid** [comment]
You're welcome ! I will continue to deep dive into Keras/Hyperas/Tensorflow for the next 6 months so i will continue to post and ask question around ! Keep up the good work!

**Comment by jesuisnicolasdavid** [comment]
So, i did get that `{{choice([Dropout(0.5), Activation('linear')])}}` is equal to a choice between Dropout or nothing. (it may be interesting to comment it in your code to explain the case).

But i did not get your example with the "three" and "four" abstraction with the conditional function though. Can you show me in a small random example how to use it in a more concrete way ? Like how to initialize the conditional function ?

```
    # If we choose 'four', add an additional fourth layer
    if conditional({{choice(['three', 'four'])}}) == 'four':
        model.add(Dense(100))
        model.add({{choice([Dropout(0.5), Activation('linear')])}})

```

**Comment by maxpumperla** [comment]
conditional is just a wrapper, it does nothing. See this for an example using this:

https://github.com/maxpumperla/hyperas/blob/master/examples/complex.py
